test/async-await/async5.js

```diff
@@ -0,0 +1,57 @@
+var util = require('util');
+var http = require('http');
+
+var test = require('../../');
+
+test('async5', async function myTest(t) {
+    try {
+        t.ok(true, 'before server');
+
+        var mockDb = { state: 'old' };
+        var server = http.createServer(function (req, res) {
+            res.end('OK');
+
+            // Pretend we write to the DB and it takes time.
+            setTimeout(function () {
+                mockDb.state = 'new';
+            }, 10);
+        });
+
+        await util.promisify(function (cb) {
+            server.listen(0, cb);
+        })();
+
+        t.ok(true, 'after server');
+
+        t.ok(true, 'before request');
+
+        var res = await util.promisify(function (cb) {
+            var req = http.request({
+                hostname: 'localhost',
+                port: server.address().port,
+                path: '/',
+                method: 'GET'
+            }, function (res) {
+                cb(null, res);
+            });
+            req.end();
+        })();
+
+        t.ok(true, 'after request');
+
+        res.resume();
+        t.equal(res.statusCode, 200);
+
+        setTimeout(function () {
```

Comment:
note that using `setTimeout` in a world of async/await is a bug. This should be `await`ing a `new Promise`, so that its fate is tied to the overarching promise of the async function.

Reply:
It’s a bad practice but not a bug. 

I’ve seen plenty of people mixing promises, async/await and callbacks

Adding ‘data’ listeners is the biggest example of mixing callbacks and promises.

This is a best practice we will need to educate the tape users on

Reply:
I consider those two things the same thing :-)

If you're returning a promise - which is always the case when using an `async function` - you are explicitly stating that your function resolves to a single future async value. Thus, every single async action you create needs to be tied to the fate of that promise.

If that's not the case, take a callback and don't be an async function/don't return a promise.

I agree educating tape users might be needed.

Reply:
100% agreed, mixing callbacks & promises is a terrible idea.

But I suspect our users might do it none the less and tape already "supports" async functions if you call `t.end()` and happen to not throw an exception so we might break existing tests